File: import-users.py
#!/usr/bin/env python3
import json
import subprocess
import sys
from pathlib import Path

def import_users():
    """Import users from JSON to new Firebase project using gcloud CLI"""
    
    json_file = Path('users-export.json')
    if not json_file.exists():
        print('❌ users-export.json not found. Run export-users.js first.')
        sys.exit(1)
    
    print('🔄 Importing users using gcloud CLI...\n')
    
    # Load users
    with open(json_file) as f:
        users = json.load(f)
    
    print(f'📥 Loaded {len(users)} users\n')
    print('⚙️  Creating users in feep-annual-event-stable-jan26...\n')
    
    results = {'ok': 0, 'exists': 0, 'failed': 0}
    created_emails = []
    
    for i, user in enumerate(users, 1):
        email = user['email']
        display_name = user['displayName'] or ''
        progress = f'[{i}/{len(users)}]'
        
        try:
            # Generate temp password
            import random, string
            temp_pass = ''.join(random.choices(string.ascii_letters + string.digits, k=16))
            
            # Create user using gcloud
            cmd = [
                '/Users/adwaitparchure/Adwait Work/google-cloud-sdk/bin/gcloud',
                'identity-toolkit', 'accounts', 'create',
                f'--email={email}',
                f'--password={temp_pass}',
                f'--display-name={display_name}',
                '--project=feep-annual-event-stable-jan26'
            ]
            
            result = subprocess.run(cmd, capture_output=True, text=True)
            
            if result.returncode == 0:
                results['ok'] += 1
                created_emails.append(email)
                print(f'{progress} ✓ {email}')
            elif 'already exists' in result.stderr:
                results['exists'] += 1
                print(f'{progress} ⚠️  {email} (already exists)')
            else:
                results['failed'] += 1
                print(f'{progress} ✗ {email} - {result.stderr[:100]}')
        except Exception as e:
            results['failed'] += 1
            print(f'{progress} ✗ {email} - {str(e)[:100]}')
    
    # Print summary
    print('\n' + '='*60)
    print('📊 IMPORT SUMMARY')
    print('='*60)
    print(f"✅ Created: {results['ok']}")
    print(f"⚠️  Already exist: {results['exists']}")
    print(f"❌ Failed: {results['failed']}")
    print('='*60)
    
    if results['ok'] > 0:
        print(f'\n✨ Successfully created {results["ok"]} users!')
        print('\n📝 Users with temporary passwords (need reset on first login):')
        for email in created_emails[:10]:
            print(f'  • {email}')
        if len(created_emails) > 10:
            print(f'  ... and {len(created_emails) - 10} more')

if __name__ == '__main__':
    import_users()

File: import-users.js
import admin from 'firebase-admin';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

// Load new service account key
const newServiceAccount = JSON.parse(fs.readFileSync('./serviceAccountKey-new.json', 'utf8'));

async function importUsers() {
  try {
    console.log('🔄 Importing users to new Firebase project...\n');
    
    // Load exported users
    const usersPath = path.join(__dirname, 'users-export.json');
    if (!fs.existsSync(usersPath)) {
      throw new Error('users-export.json not found. Run export-users.js first.');
    }

    const users = JSON.parse(fs.readFileSync(usersPath, 'utf8'));
    console.log(`📥 Loaded ${users.length} users from export\n`);

    // Initialize new project - use default app
    if (admin.apps.length > 0) {
      await admin.app().delete();
    }

    admin.initializeApp({
      credential: admin.credential.cert(newServiceAccount)
    });

    const auth = admin.auth();
    const results = [];

    // Import users
    console.log('⚙️  Creating users in new Firebase project...\n');
    for (let i = 0; i < users.length; i++) {
      const user = users[i];
      const progress = `[${i + 1}/${users.length}]`;

      try {
        // Generate a temporary random password
        const tempPassword = Math.random().toString(36).slice(-16);

        const userData = {
          email: user.email,
          password: tempPassword,
          disabled: false
        };

        if (user.displayName) userData.displayName = user.displayName;
        if (user.photoURL && user.photoURL.startsWith('http')) userData.photoURL = user.photoURL;

        // Create user
        const newUser = await auth.createUser(userData);

        results.push({
          status: 'OK',
          email: user.email,
          oldUid: user.uid,
          newUid: newUser.uid,
          error: ''
        });

        console.log(`${progress} ✓ ${user.email}`);
      } catch (err) {
        // Check if email already exists
        if (err.code === 'auth/email-already-exists') {
          results.push({
            status: 'EXISTS',
            email: user.email,
            oldUid: user.uid,
            newUid: '',
            error: 'Email already exists in new project'
          });
          console.log(`${progress} ⚠️  ${user.email} (already exists)`);
        } else {
          results.push({
            status: 'FAIL',
            email: user.email,
            oldUid: user.uid,
            newUid: '',
            error: err.message
          });
          console.log(`${progress} ✗ ${user.email} - ${err.message}`);
        }
      }
    }

    // Save results
    const csvPath = path.join(__dirname, 'import-results.csv');
    const csvHeader = 'Status,Email,Old UID,New UID,Error\n';
    const csvRows = results.map(r => 
      `${r.status},"${r.email}","${r.oldUid}","${r.newUid}","${r.error}"`
    ).join('\n');
    fs.writeFileSync(csvPath, csvHeader + csvRows);

    const ok = results.filter(r => r.status === 'OK').length;
    const exists = results.filter(r => r.status === 'EXISTS').length;
    const fail = results.filter(r => r.status === 'FAIL').length;

    console.log('\n' + '='.repeat(60));
    console.log('📊 IMPORT SUMMARY');
    console.log('='.repeat(60));
    console.log(`✅ Created: ${ok}`);
    console.log(`⚠️  Already exist: ${exists}`);
    console.log(`❌ Failed: ${fail}`);
    console.log(`📄 Results: ${csvPath}`);
    console.log('='.repeat(60));

    if (ok > 0) {
      console.log('\n✨ User import completed!');
      console.log(`\n📝 IMPORTANT NOTES:`);
      console.log(`• ${ok} users created with temporary passwords`);
      console.log(`• Users must reset password on first login`);
      console.log(`• Send password reset emails to users:\n`);

      // List emails for reset
      const createdEmails = results.filter(r => r.status === 'OK').map(r => r.email);
      console.log(createdEmails.join('\n'));
    }

    process.exit(fail > 0 ? 1 : 0);
  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  }
}

importUsers();
